radix-engine/src/blueprints/pool/two_resource_pool/blueprint.rs

```diff
@@ -436,20 +436,17 @@ impl TwoResourcePoolBlueprint {
                 (false, false, false) => Ok((
                     contribution1
                         .safe_mul(contribution2)
-                        .unwrap()
-                        .sqrt()
-                        .unwrap(),
+                        .and_then(|d| d.sqrt())
+                        .ok_or_else(|| TwoResourcePoolError::DecimalOverflowError)?,
                     contribution1,
                     contribution2,
                 )),
                 (false, _, _) => Ok((
```

Comment:
In a similar manner to the case above, I believe that we should expand out the roots for this to be:

```math
mint\_amount = \sqrt{(contribution_1 + reserves_1) * (contribution_2 + reserves_2)} = \sqrt{(contribution_1 + reserves_1)} * \sqrt{(contribution_2 + reserves_2)}
```

Where $`contribution_n + reserves_n`$ could overflow.